low-level/files/opt/seagate/sspl/bin/generate_sspl_bundle/sspl_bundle_generate

```diff
@@ -325,8 +328,27 @@ def sspl_bundle_cleanup():
     if os.path.exists(sspl_bundle_tmp):
         shutil.rmtree(sspl_bundle_tmp)
 
+def validate_sb_request():
+    valid_request = True
+    if os.path.isfile(onGoingRequestPath):
+        last_reboot = int(psutil.boot_time())
+        req_start_time = int(pathlib.Path(onGoingRequestPath).stat().st_mtime)
+        current_time = int(time.time())
+        # Check if system is rebooted in between or request is expired.
+        # If Yes, Then clear existing request cache.
+        is_req_expired = (current_time - req_start_time) > SB_REQUEST_TIMEOUT
```

Comment:
Here we are invalidating our previous support bundle request after x amount of time, but we can also rather ensure the cleanup process by subscribing to the SIGTERM event.

Reply:
Yes. but here we are cleaning temp files of support bundle which will be restricting new sb request while existing request is processing.

Reply:
Yes but what if I start the generation, and it gets terminated due to some reason, now for next 20 mins I can not restart it.

Reply:
yes. we are cleaning up the things when the program terminates by itself. if 3rd person forcefully terminates it then we need to wait for the next 20 min. but that is an exceptional case that no one forcefully terminates the program.

Reply:
correct me if I am wrong, but SIGTERM signal handling works on forceful termination also.
But if it is okay to wait 20 mins then we are good 👍

Reply:
We are handling all this scenario inside the sb script only. and this script directly calls by 3rd application. that SIGTERM signal will be received outside the script where we run the script. I don't see we can manage this logic inside the sb script itself. those will be taking care of by which script is calling sb script.

https://unix.stackexchange.com/a/10327